File: app/src/main/java/com/dave/mainactivity/network/BooksService.java
package com.dave.mainactivity.network;

import com.dave.mainactivity.model.BooksResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Url;

public interface BooksService {
    @GET
    Call<BooksResponse> getListOfBooks(@Url String str);
}

File: app/src/main/java/com/dave/mainactivity/service/BooksService.kt
package com.dave.mainactivity.service

import com.dave.mainactivity.enums.SearchType
import com.dave.mainactivity.model.Book

interface BooksService {
    fun getBooksList(input: String, searchType: SearchType = SearchType.all, closure: (List<Book>?) -> Unit)
}
